[Search][Playground] Enable Editable Query Mode for Chat

TattdCodeMonkey opened the pull request:
## Summary

- Enable the updated Playground Query mode for chat without feature flag
- Added telemetry event for when the elasticsearch query is edited by the user.

### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed

TattdCodeMonkey committed:
playground: add telemetry event for editing es query

TattdCodeMonkey committed:
playground: ungate chat new query mode

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/298793)
* Commit: b9c3204ac4a3255aa2f5b2f13930f88250e526c6

### Failed CI Steps
* [FTR Configs #12](https://buildkite.com/elastic/kibana-pull-request/builds/298793#0196a279-f6dd-496f-b43f-6304fe0010f0)
* [FTR Configs #12](https://buildkite.com/elastic/kibana-pull-request/builds/298793#0196a2a2-52d7-4f62-9d57-5f46a387bdf3)
* [FTR Configs #101](https://buildkite.com/elastic/kibana-pull-request/builds/298793#0196a279-f72b-478b-8bf0-94a48563b8db)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298793#0196a279-f72b-478b-8bf0-94a48563b8db) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298793/jobs/0196a279-f72b-478b-8bf0-94a48563b8db/artifacts/0196a2a6-7d72-447f-91bb-ff7b526834ca) FTR Configs #<span></span>101 / Security Solution -  Telemetry Security Telemetry - Indices metadata task telemetry @<span></span>ess @<span></span>serverless indices metadata should publish index stats events
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298793#0196a279-f6dd-496f-b43f-6304fe0010f0) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298793/jobs/0196a279-f6dd-496f-b43f-6304fe0010f0/artifacts/0196a2a2-254c-4ec5-a292-45b77d5e9388) FTR Configs #<span></span>12 / serverless search UI Serverless Playground Overview chat page chat show fields and code in view query
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/298793#0196a2a2-52d7-4f62-9d57-5f46a387bdf3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/298793/jobs/0196a2a2-52d7-4f62-9d57-5f46a387bdf3/artifacts/0196a2b6-c59c-4a75-889b-a1bd82d391d2) FTR Configs #<span></span>12 / serverless search UI Serverless Playground Overview chat page chat show fields and code in view query

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/298793","number":298793,"commit":"b9c3204ac4a3255aa2f5b2f13930f88250e526c6"}],"number":298793}
buildkite-pr-comment-->

TattdCodeMonkey committed:
test(playground): get code from code editor

TattdCodeMonkey committed:
test(playground): remove Promise.race from llm proxy

Implemented the timeout error in a single promise instead of using
Promise.race(). In testing I was seeing periodic crashes of the test
runner because the timeout Promise rejection was unhandled. By
consolidating to a single promise this should allow the tests to not
leave any dangling promises.

TattdCodeMonkey committed:
test(playground): test new query mode features

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/299170)
* Commit: e9306caa0ceeb49fe9438e366a9a26606559786b

### Failed CI Steps
* [Linting (with types)](https://buildkite.com/elastic/kibana-pull-request/builds/299170#0196a6c4-8850-4192-b57d-fbf75192ae3e)

### History
* :green_heart: [Build #299138](https://buildkite.com/elastic/kibana-pull-request/builds/299138) succeeded 18a31949189d1ff85419136f303298d9715bbfc4
* :broken_heart: [Build #298793](https://buildkite.com/elastic/kibana-pull-request/builds/298793) failed b9c3204ac4a3255aa2f5b2f13930f88250e526c6

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299170","number":299170,"commit":"e9306caa0ceeb49fe9438e366a9a26606559786b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299138","number":299138,"commit":"18a31949189d1ff85419136f303298d9715bbfc4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/298793","number":298793,"commit":"b9c3204ac4a3255aa2f5b2f13930f88250e526c6"}],"number":299170}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/299175)
* Commit: 31c6b55ab4cc5de5813a234a195ab14c5ef8a89a

### Failed CI Steps
* [Linting (with types)](https://buildkite.com/elastic/kibana-pull-request/builds/299175#0196a6df-3ad7-4f02-969b-dbfa9bf95bd8)

### History
* :broken_heart: [Build #299170](https://buildkite.com/elastic/kibana-pull-request/builds/299170) failed e9306caa0ceeb49fe9438e366a9a26606559786b
* :green_heart: [Build #299138](https://buildkite.com/elastic/kibana-pull-request/builds/299138) succeeded 18a31949189d1ff85419136f303298d9715bbfc4
* :broken_heart: [Build #298793](https://buildkite.com/elastic/kibana-pull-request/builds/298793) failed b9c3204ac4a3255aa2f5b2f13930f88250e526c6

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299175","number":299175,"commit":"31c6b55ab4cc5de5813a234a195ab14c5ef8a89a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299170","number":299170,"commit":"e9306caa0ceeb49fe9438e366a9a26606559786b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299138","number":299138,"commit":"18a31949189d1ff85419136f303298d9715bbfc4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/298793","number":298793,"commit":"b9c3204ac4a3255aa2f5b2f13930f88250e526c6"}],"number":299175}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/299192)
* Commit: f0b6b70fd932924e581420878d1bf351aa6f828e

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/299192#0196a729-76dd-4247-82eb-e5d7b369e24c)

### History
* :broken_heart: [Build #299175](https://buildkite.com/elastic/kibana-pull-request/builds/299175) failed 31c6b55ab4cc5de5813a234a195ab14c5ef8a89a
* :broken_heart: [Build #299170](https://buildkite.com/elastic/kibana-pull-request/builds/299170) failed e9306caa0ceeb49fe9438e366a9a26606559786b
* :green_heart: [Build #299138](https://buildkite.com/elastic/kibana-pull-request/builds/299138) succeeded 18a31949189d1ff85419136f303298d9715bbfc4
* :broken_heart: [Build #298793](https://buildkite.com/elastic/kibana-pull-request/builds/298793) failed b9c3204ac4a3255aa2f5b2f13930f88250e526c6

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299192","number":299192,"commit":"f0b6b70fd932924e581420878d1bf351aa6f828e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299175","number":299175,"commit":"31c6b55ab4cc5de5813a234a195ab14c5ef8a89a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299170","number":299170,"commit":"e9306caa0ceeb49fe9438e366a9a26606559786b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299138","number":299138,"commit":"18a31949189d1ff85419136f303298d9715bbfc4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/298793","number":298793,"commit":"b9c3204ac4a3255aa2f5b2f13930f88250e526c6"}],"number":299192}
buildkite-pr-comment-->

TattdCodeMonkey committed:
test(playground): updated expect for query mode run

TattdCodeMonkey committed:
test(playground): type fix in llm proxy

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/299209)
* Commit: ccdc799ff3a4d9f865e9d8fe4923b04f08fe5bdd

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/299209#0196a75e-a622-46fe-b940-69c5da34bdf4)

### History
* :broken_heart: [Build #299199](https://buildkite.com/elastic/kibana-pull-request/builds/299199) failed e5bb1b38514e43a762a4f627280634772c6baa9c
* :broken_heart: [Build #299192](https://buildkite.com/elastic/kibana-pull-request/builds/299192) failed f0b6b70fd932924e581420878d1bf351aa6f828e
* :broken_heart: [Build #299175](https://buildkite.com/elastic/kibana-pull-request/builds/299175) failed 31c6b55ab4cc5de5813a234a195ab14c5ef8a89a
* :broken_heart: [Build #299170](https://buildkite.com/elastic/kibana-pull-request/builds/299170) failed e9306caa0ceeb49fe9438e366a9a26606559786b
* :green_heart: [Build #299138](https://buildkite.com/elastic/kibana-pull-request/builds/299138) succeeded 18a31949189d1ff85419136f303298d9715bbfc4
* :broken_heart: [Build #298793](https://buildkite.com/elastic/kibana-pull-request/builds/298793) failed b9c3204ac4a3255aa2f5b2f13930f88250e526c6

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299209","number":299209,"commit":"ccdc799ff3a4d9f865e9d8fe4923b04f08fe5bdd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299199","number":299199,"commit":"e5bb1b38514e43a762a4f627280634772c6baa9c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299192","number":299192,"commit":"f0b6b70fd932924e581420878d1bf351aa6f828e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299175","number":299175,"commit":"31c6b55ab4cc5de5813a234a195ab14c5ef8a89a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299170","number":299170,"commit":"e9306caa0ceeb49fe9438e366a9a26606559786b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299138","number":299138,"commit":"18a31949189d1ff85419136f303298d9715bbfc4"}],"number":299209}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/299217)
* Commit: c0d25464c32971f686472627d7e85e11b863872a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/81385f6c53153aa1adc0a64cc0663798574e889e) | [after](https://github.com/elastic/kibana/commit/c0d25464c32971f686472627d7e85e11b863872a) | diff |
| --- | --- | --- | --- |
| `searchPlayground` | 337 | 336 | -1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/81385f6c53153aa1adc0a64cc0663798574e889e) | [after](https://github.com/elastic/kibana/commit/c0d25464c32971f686472627d7e85e11b863872a) | diff |
| --- | --- | --- | --- |
| `searchPlayground` | 209.9KB | 207.4KB | -2.6KB |



### History
* :broken_heart: [Build #299209](https://buildkite.com/elastic/kibana-pull-request/builds/299209) failed ccdc799ff3a4d9f865e9d8fe4923b04f08fe5bdd
* :broken_heart: [Build #299199](https://buildkite.com/elastic/kibana-pull-request/builds/299199) failed e5bb1b38514e43a762a4f627280634772c6baa9c
* :broken_heart: [Build #299192](https://buildkite.com/elastic/kibana-pull-request/builds/299192) failed f0b6b70fd932924e581420878d1bf351aa6f828e
* :broken_heart: [Build #299175](https://buildkite.com/elastic/kibana-pull-request/builds/299175) failed 31c6b55ab4cc5de5813a234a195ab14c5ef8a89a
* :broken_heart: [Build #299170](https://buildkite.com/elastic/kibana-pull-request/builds/299170) failed e9306caa0ceeb49fe9438e366a9a26606559786b
* :green_heart: [Build #299138](https://buildkite.com/elastic/kibana-pull-request/builds/299138) succeeded 18a31949189d1ff85419136f303298d9715bbfc4

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299217","number":299217,"commit":"c0d25464c32971f686472627d7e85e11b863872a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299209","number":299209,"commit":"ccdc799ff3a4d9f865e9d8fe4923b04f08fe5bdd"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299199","number":299199,"commit":"e5bb1b38514e43a762a4f627280634772c6baa9c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299192","number":299192,"commit":"f0b6b70fd932924e581420878d1bf351aa6f828e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299175","number":299175,"commit":"31c6b55ab4cc5de5813a234a195ab14c5ef8a89a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/299170","number":299170,"commit":"e9306caa0ceeb49fe9438e366a9a26606559786b"}],"number":299217}
buildkite-pr-comment-->

saarikabhasi commented on x-pack/solutions/search/plugins/search_playground/public/components/query_mode/query_viewer.tsx:
```diff
@@ -37,6 +39,8 @@ export const ElasticsearchQueryViewer = ({
 }) => {
   const { euiTheme } = useEuiTheme();
+  const usageTracker = useUsageTracker();
+  const [esQueryFirstEdit, setEsQueryFirtEdit] = useState<boolean>(false);
```
nit: typo in `setEsQueryFirtEdit` to `setEsQueryFirstEdit`

saarikabhasi reviewed (approved):
LGTM. Other than tiny nit change.

saarikabhasi commented on x-pack/solutions/search/plugins/search_playground/public/components/playgorund.tsx:
unrelated to this change. but we need to fix typo in filename too

TattdCodeMonkey replied:
agreed

TattdCodeMonkey commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into playground/enabled-chat-edit-query

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8249](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8249)
[✅] x-pack/test/functional/apps/search_playground/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F220158%2Fhead)

TattdCodeMonkey commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into playground/enabled-chat-edit-query
